extensions/kafka/src/main/java/io/deephaven/kafka/KeyOrValueSpecObjectProcessorImpl.java

```diff
@@ -0,0 +1,129 @@
+package io.deephaven.kafka;
+
+import io.confluent.kafka.schemaregistry.SchemaProvider;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.deephaven.chunk.ObjectChunk;
+import io.deephaven.chunk.WritableChunk;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.engine.table.ColumnDefinition;
+import io.deephaven.engine.table.TableDefinition;
+import io.deephaven.kafka.KafkaTools.Consume.KeyOrValueSpec;
+import io.deephaven.kafka.KafkaTools.KeyOrValue;
+import io.deephaven.kafka.KafkaTools.KeyOrValueIngestData;
+import io.deephaven.kafka.ingest.KafkaStreamPublisher;
+import io.deephaven.kafka.ingest.KeyOrValueProcessor;
+import io.deephaven.kafka.ingest.MultiFieldChunkAdapter;
+import io.deephaven.processor.ObjectProcessor;
+import io.deephaven.qst.type.Type;
+import org.apache.commons.lang3.mutable.MutableInt;
+import org.apache.kafka.common.serialization.Deserializer;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * This implementation is useful for presenting an easier onboarding ramp and better (and public) interface
+ * {@link KafkaTools.Consume#objectProcessorSpec(Deserializer, ObjectProcessor, List)} for end-users. The
+ * {@link ObjectProcessor} is a user-visible replacement for {@link KeyOrValueProcessor}. In the meantime though, we are
+ * adapting into a {@link KeyOrValueProcessor} until such a time when {@link KafkaStreamPublisher} can be re-written to
+ * take advantage of these better interfaces.
+ */
+class KeyOrValueSpecObjectProcessorImpl<T> extends KeyOrValueSpec {
+    private final Deserializer<? extends T> deserializer;
+    private final ObjectProcessor<? super T> processor;
+    private final List<String> columnNames;
+
+    KeyOrValueSpecObjectProcessorImpl(
+            Deserializer<? extends T> deserializer, ObjectProcessor<? super T> processor, List<String> columnNames) {
+        if (columnNames.size() != processor.outputTypes().size()) {
```

Comment:
So; your own processor implementation _can_ know about the column names, but I don't know if it makes sense to elevate that to the interface level. While it would simplify some calls sites (like this constructor), I think there are legitimate use cases where it won't make sense and would limit compose-ability.

As such, I _think_ the interface should be based around the minimal information needed for `void processAll(ObjectChunk<? extends T, ?> in, List<WritableChunk<?>> out)` to work.